src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,155 @@
+package frc.trigon.robot.subsystems.arm;
+
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.trigon.robot.utilities.Conversions;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+    private final CANSparkMax
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            masterAngleMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerAngleMotor = ArmConstants.MASTER_ELEVATOR_MOTOR;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGeneration,
+            lastElevatorMotorProfileGeneration;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    void generateElevatorMotorProfile(double targetElevatorPositionMeters, double elevatorSpeedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, elevatorSpeedPercentage),
+                new TrapezoidProfile.State(targetElevatorPositionMeters, 0),
+                new TrapezoidProfile.State(getElevatorPositionMeters(), getElevatorVelocityMetersPerSeconds())
+        );
+
+        lastElevatorMotorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle, double angleSpeedPercentage) {
```

Comment:
rename the param here to speedPercentage since this is for the angle by the function name already